### ISSUE TITLE
YAM should use "Content-Disposition: attachment" for all attachments

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#507]
**Originally on 2014-01-13 17:00:40 +0100**

___
## Summary
YAM does currently automatically decide if a plain text attachment will be signaled to be "inlined" or being specified as being an "attachment" by using the "Content-Disposition:" header parameter of a MIME boundary. While this seems to be perfectly valid according to the RFCs, it would be better to always use "attachment" to specify that even a plain text attachment which the user manually added should be an "attachment" rather than just inlining the text under a main letter part. 

## Steps to reproduce
 1. generate a plain text file and save it as "test.txt"
 2. attach it to an example mail and store it to a folder
 3. have a look at the raw mail. YAM will use "Content-Disposition: inline" because it identifies it as plain text
 4. send this mail to someone with an "Apple Mail" client or an iOS device and note that it will not be displayed as an actual attachment but the text of that plain text attachment will directly display under the main letter part

## Expected results
YAM should send manually added attachment with "Content-Disposition: attachment" as otherwise these mails might not be accessible as actual attachments by other mail clients.

## Actual results
With such mails there is no possibility to save/export/strip such plain text parts in iOS mail clients or in Apple Mail.

**Comment by tboeckel** [comment]
**Originally on 2014-01-14 09:43:40 +0100**

___
In (29fcc61):
 * YAM_WR.c, mui/WriteWindow.c: attachments are now always flagged as such in the generated mail instead of letting YAM choose their content disposition on basis of their type (text or not). This closes #507.

**Comment by jens-maus** [comment]
**Originally on 2014-01-14 11:27:23 +0100**

___
In (05ac6cd):
 - make sure that "Content-Disposition: inline" will only be used if there is
  really a printable part. Use "attachment" otherwise. This refs #507.

**Comment by jens-maus** [comment]
**Originally on 2014-01-14 15:07:57 +0100**

___
I just found out that the "Content-Disposition:" sub-header seems to be completely missing for attached emails. See here:

```
----=_BOUNDARY.27eff86845557eb6.92
Content-Type: message/rfc822
Content-Description: Test

Return-Path: test@test.de
[...]
```

So the only two headers emitted are Content-Type and Content-Description if a user forwards an email "as attachment" using the menu option in YAM.

See here, for example, what Apple Mail emits for the same functionality:

```
--Apple-Mail=_18472E10-8F4F-4251-BE2E-2140FF5A94DF
Content-Transfer-Encoding: 7bit
Content-Disposition: attachment;
	filename=Abwesenheitsvertretung.eml
Content-Type: message/rfc822;
	x-mac-hide-extension=yes;
	name="Abwesenheitsvertretung.eml"

[...]
```

So it seems we should at least add the "Content-Disposition: attachment" header as well as specifying a "filename=" parameter which should correspond to the "subject+.eml"  pattern.

**Comment by tboeckel** [comment]
**Originally on 2014-01-15 08:43:14 +0100**

___
In (7dea5eb):
 * YAM_WR.c: mails being forwarded as attachment now get a proper name. This in turn lets YAM generate some more header lines for the attachment part. This closes #507 again.

**Comment by jens-maus** [comment]
**Originally on 2014-04-07 22:39:49 +0200**

___
**Release Note:**
fixed a problem with the "Content-Disposition:" MIME header line which was set to 'inline' for plain text attachments. This cause problems with some mail clients (e.g. Apple Mail) which always displayed these attachments as part of the mail text not allowing to save them as separate attachments.
